test/wappsto_test.py

```diff
@@ -1,672 +1,247 @@
-    def test_instantiation_device_list_2_devices_passes(self):
-
-        assert len(wappsto.Wappsto(json_file_name=self.json_ok_two_devices).instance.device_list) == 2
-
-    def test_instantiation_parsing_fails(self):
-
-        with pytest.raises(json.decoder.JSONDecodeError):
-            wappsto.Wappsto(json_file_name=self.json_incorrect)
-
-    def test_instantiation_parsing_invalid_JSON_errors(self):
-        with pytest.raises(Exception):
-            wappsto.Wappsto(json_file_name=self.json_incorrect)
-
-    def test_instantiation_parsing_file_not_found_errors(self):
-
-        with pytest.raises(FileNotFoundError):
-            wappsto.Wappsto(json_file_name="blabla")
-
-
-class TestDevice:
-
-    @classmethod
+        self.test_json_prettyprint_location = os.path.join(os.path.dirname(__file__), TEST_JSON_prettyprint)
+        self.test_json_location = os.path.join(os.path.dirname(__file__), TEST_JSON)
+    
+    def test_load_prettyprint_json(self):
+        #Arrange
+        with open(self.test_json_location,"r") as json_file:
+            decoded = json.load(json_file)
+        
+        #Act
+        service = wappsto.Wappsto(json_file_name=self.test_json_prettyprint_location)
+        
+        #Assert
+        assert service.instance.decoded == decoded
+
+class TestConnClass:
+    
     def setup_method(self):
-        self.json_ok_two_devices = os.path.join(os.path.dirname(__file__), 'test_JSON/e9ec36e2-0bd5-4fef-8e7b-9c200e93ebce.json')
-        self.json_incorrect = os.path.join(os.path.dirname(__file__), 'test_JSON/e9ec36e2-0bd5-4fef-8e7b-9c200e93ebce_invalid.json')
-        self.wapp = wappsto.Wappsto(json_file_name=self.json_ok_two_devices)
-
-    def test_finding_device_passes(self):
-        device = self.wapp.get_device("test_device_all_fieldsasdasd")
-        assert device is not None
-
-    def test_finding_device_errors(self):
-        self.test_wapp = wappsto.Wappsto(json_file_name=self.json_ok_two_devices)
-        with pytest.raises(Exception):
-            self.test_wapp.get_device("asdfasf")
-
-    def test_set_value_callback_passes(self):
-        device = self.wapp.get_device("test_device_all_fieldsasdasd")
-        value = device.get_value("test_value_with_number")
-        def value_callback(self):
+        test_json_location = os.path.join(os.path.dirname(__file__), TEST_JSON)
+        self.service = wappsto.Wappsto(json_file_name=test_json_location)
+        self.service.RETRY_LIMIT = 1
+    
+    @pytest.mark.parametrize("address,port,expected_status", [(ADDRESS,PORT,status.RUNNING),
+                                                     (ADDRESS,-1,status.DISCONNECTING),
+                                                     ("wappstoFail.com",PORT,status.DISCONNECTING)])
+    def test_connection(self,address,port,expected_status):
+        #Arrange
+        
+        #Act
+        try:
+            with patch('time.sleep', return_value=None):
+                self.service.start(address=address, port=port)
```

Comment:
Should you not also mock the socket here? Because you should never connect to the "real" world.